Add a new button to import an existing layers from other plugins

adamltyson opened the pull request:
## Description
There's now a good number of segmentation plugins in napari. Following a conversation with @SaraMederos, it seems like a good idea to be able to "import" these results into `brainreg-segment` so that they can be analysed in the context of the atlas.
**What is this PR**

- [ ] Bug fix
- [x] Addition of a new feature
- [ ] Other

**What does this PR do?**
* Adds a `Add region from selected layer` to the region segmentation panel that will add the selected layer (e.g. from a segmentation plugin) to the list of layers to be analysed.

* Adds a `Add track from selected layer` to the track segmentation panel that will add the selected layer (e.g. from another plugin) to the list of layers to be analysed.

## References
Closes #73 

## How has this PR been tested?

No automated tests yet, as all the tests are broken in some way. 

## Is this a breaking change?

No

## Does this PR require an update to the documentation?
Yes, it will, not added yet though.

## Checklist:

- [x] The code has been tested locally
- [x] Tests have been added to cover all new functionality (unit & integration)
- [x] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

adamltyson commented:
Converted to draft as I'd like to get some feedback on the tool from a user POV, and also add similar functionality for tracks.

adamltyson commented:
Also need to add tests

adamltyson commented:
Moved the test changes from this branch into #77

dstansby commented on brainreg_segment/segmentation_panels/regions.py:
```diff
@@ -144,6 +156,28 @@ def add_region(self):
+                )
+        else:
+            if confirmation:
+                display_info(
```
Are these confirmation boxes meant to be turned on by default? If I try and add a new region from a points layer, I don't get an info box popping up saying that the layer isn't a labels layer.

adamltyson replied:
Fixed in edbf298c76d09d8e987c04dfbe143489008da0e6

dstansby commented on brainreg_segment/regions/layers.py:
```diff
@@ -74,6 +74,13 @@ def add_existing_label_layers(
     return label_layer
 
 
+def add_region_from_existing_layer(selected_layer, label_layers):
+    """
+    Adds an existing label layer (e.g. from another plugin) to the list
+    """
+    label_layers.append(selected_layer)
```
```suggestion
def add_region_from_existing_layer(new_layer, label_layers):
    """
    Adds an existing label layer (e.g. from another plugin) to the list
    """
    label_layers.append(new_layer)
```
since in this context it's not neccesarily a 'selected' layer being added.

dstansby commented on brainreg_segment/regions/layers.py:
```diff
@@ -74,6 +74,13 @@ def add_existing_label_layers(
     return label_layer
 
 
+def add_region_from_existing_layer(selected_layer, label_layers):
+    """
+    Adds an existing label layer (e.g. from another plugin) to the list
+    """
+    label_layers.append(selected_layer)
```
I think it might be worth having a check that the layer is a labels type here, and erroring if not? That way if this is used programattically it will error correctly. Then in `regions.py` where the gui stuff happens a `try ..., except ...` pattern can be used to catch the error and pop up a info box.

adamltyson replied:
Good idea. Added in 52d3d934d779c1375d43e55ba387fd416f84a31d.

dstansby commented on brainreg_segment/tracks/layers.py:
```diff
@@ -26,3 +25,10 @@ def add_existing_track_layers(viewer, track_file, point_size):
+    """
+    Adds an existing tracks layer (e.g. from another plugin) to the list
+    """
+    track_layers.append(selected_layer)
```
Similar to above, I think worth checking the layer type and erroring if not the right error type here.

dstansby reviewed (commented):
Three major comments:

- See inline comments, I think worth checking the layer type and erroring when adding the existing layers in the functions that do the adding.
- I can't get any of the info boxes to pop up, even when trying to add a layer that is the wrong type. Not sure what the best way to debug this is though...
- This could really do with some user facing docs, but I guess we don't have anywhere to put them right now? Perhaps a simple markdown file with a short "how to import existing layers" section could be started, and then when a proper doc structure is added to the repo it can be migrated there.

SaraMederos commented:
> For the docs, I have created a temporary markdown file to include. 
I will anyway work on a more detailed document to include later on

adamltyson commented:
Thanks @dstansby I've addressed these issues. I'll document this feature in the new docs site before the new version is released. @SaraMederos is also kindly developing a tutorial from the researchers point of view.

dstansby reviewed (approved):
I haven't checked this works as intended, but I have smoke checked it and the info boxes seem to be working now 👍
